src/AnnotationDialog.js

```diff
@@ -569,106 +546,38 @@ class AnnotationDialog extends EventEmitter {
-
+    renderAnnotations() {
         const annotationContainerEl = this.dialogEl.querySelector(`.${CLASS_COMMENTS_CONTAINER}`);
-        annotationContainerEl.appendChild(annotationEl);
-
-        // Avatar
-        const avatarEl = document.createElement('div');
-        avatarEl.classList.add(constants.CLASS_PROFILE_IMG_CONTAINER);
-        avatarEl.innerHTML = avatarHtml;
-        annotationEl.appendChild(avatarEl);
-
-        // Creator namate & date
-        const profileContainerEl = document.createElement('div');
-        profileContainerEl.classList.add(constants.CLASS_PROFILE_CONTAINER);
-        annotationEl.appendChild(profileContainerEl);
-
-        const userNameEl = document.createElement('div');
-        userNameEl.classList.add(constants.CLASS_USER_NAME);
-        userNameEl.textContent = userName;
-        profileContainerEl.appendChild(userNameEl);
-
-        const createdEl = document.createElement('div');
-        createdEl.classList.add(constants.CLASS_COMMENT_DATE);
-        createdEl.textContent = created;
-        profileContainerEl.appendChild(createdEl);
-
-        // Comment
-        const commentTextEl = document.createElement('div');
-        commentTextEl.appendChild(textEl);
-        annotationEl.appendChild(commentTextEl);
-
-        // Delete button
-        if (!annotation.permissions.can_delete) {
-            return;
-        }
-
-        const deleteBtn = util.generateBtn(
-            [constants.CLASS_BUTTON_PLAIN, CLASS_BUTTON_DELETE_COMMENT],
-            this.localized.deleteButton,
-            ICON_DELETE,
-            constants.DATA_TYPE_DELETE
-        );
-        annotationEl.appendChild(deleteBtn);
-
-        const deleteConfirmEl = document.createElement('div');
-        deleteConfirmEl.classList.add(CLASS_DELETE_CONFIRMATION);
-        deleteConfirmEl.classList.add(constants.CLASS_HIDDEN);
-        annotationEl.appendChild(deleteConfirmEl);
-
-        const confirmMsgEl = document.createElement('div');
-        confirmMsgEl.classList.add(constants.CLASS_DELETE_CONFIRM_MESSAGE);
-        confirmMsgEl.textContent = this.localized.deleteConfirmation;
-        deleteConfirmEl.appendChild(confirmMsgEl);
-
-        const deleteBtnsEl = document.createElement('div');
-        deleteBtnsEl.classList.add(constants.CLASS_BUTTON_CONTAINER);
-        deleteConfirmEl.appendChild(deleteBtnsEl);
-
-        const cancelDeleteBtn = util.generateBtn(
-            [constants.CLASS_BUTTON, CLASS_CANCEL_DELETE],
-            this.localized.cancelButton,
-            this.localized.cancelButton,
-            constants.DATA_TYPE_CANCEL_DELETE
+        const locale = this.locale.substr(0, this.locale.indexOf('-'));
+
+        this.annotationListComponent = render(
+            <ul className='ba-annotation-list'>
+                {this.annotations.map((item) => {
+                    const { annotationID, created, modified, text, user, type } = item;
+                    if (type === constants.TYPES.highlight) {
```

Comment:
Does this mean we filter out rendering any highlight annotations? Isn't there still a dialog if the highlight has comments?

Reply:
that's only for plain highlight. we also have `TYPES.highlight_comment` :)